src/components/MAPI/releases/utils.ts

```diff
@@ -0,0 +1,159 @@
+import ErrorReporter from 'lib/errors/ErrorReporter';
+import { IRelease, ReleaseHelper } from 'lib/ReleaseHelper';
+import { compare } from 'lib/semver';
+import * as releasev1alpha1 from 'model/services/mapi/releasev1alpha1';
+import { Providers } from 'shared/constants';
+import { PropertiesOf } from 'shared/types';
+import * as ui from 'UI/Display/MAPI/releases/types';
+
+export function getReleaseHelper(
+  currVersion: string,
+  provider: PropertiesOf<typeof Providers>,
+  isAdmin: boolean,
+  releases: releasev1alpha1.IRelease[],
+  ignorePreReleases = true
+) {
+  const mappedReleases = releases.reduce(
+    (acc: Record<string, IRelease>, r: releasev1alpha1.IRelease) => {
+      // Remove the `v` prefix.
+      const normalizedVersion = r.metadata.name.slice(1);
+
+      acc[normalizedVersion] = {
+        version: normalizedVersion,
+        active: r.spec.state === 'active',
+      };
+
+      return acc;
+    },
+    {}
+  );
+
+  const releaseHelper = new ReleaseHelper({
+    availableReleases: mappedReleases,
+    provider,
+    currentReleaseVersion: currVersion,
+    isAdmin,
+    ignorePreReleases,
+  });
+
+  return releaseHelper;
+}
+
+export function getSupportedUpgradeVersions(
+  currVersion: string,
+  provider: PropertiesOf<typeof Providers>,
+  isAdmin: boolean,
+  releases: releasev1alpha1.IRelease[]
+): ui.IReleaseVersion[] {
+  try {
+    const releaseHelper = getReleaseHelper(
+      currVersion,
+      provider,
+      isAdmin,
+      releases
+    );
+
+    const availableReleases = releaseHelper.getSupportedUpgradeVersions();
+
+    return availableReleases.map((r) => {
+      const status =
+        r.getPreRelease().length > 0
+          ? ui.ReleaseVersionStatus.PreRelease
+          : ui.ReleaseVersionStatus.Stable;
+
+      return {
+        version: r.toString(),
+        status,
+      };
+    });
+  } catch (err) {
+    ErrorReporter.getInstance().notify(err);
+
+    return [];
+  }
+}
+
+interface IComponent {
+  name: string;
+  version: string;
+}
+
+function reduceReleaseToComponents(
+  release: releasev1alpha1.IRelease
+): Record<string, IComponent> {
+  const components: Record<string, IComponent> = {};
+
+  for (const component of release.spec.components) {
+    components[component.name] = {
+      name: component.name,
+      version: component.version,
+    };
+  }
+
+  if (release.spec.apps) {
+    for (const component of release.spec.apps) {
+      components[component.name] = {
+        name: component.name,
+        version: component.version,
+      };
+    }
+  }
+
+  return components;
+}
+
+export function getReleaseComponentsDiff(
```

Comment:
Created an utility for getting the diff between the components in 2 releases, instead of using the `diff` external library that we currently use in the GS API implementation.